agent/dns.go

```diff
@@ -793,32 +794,44 @@ func (d *DNSServer) doDispatch(network string, remoteAddr net.Addr, req, resp *d
 			if err != nil {
 				return invalid()
 			}
-
-			resp.Answer = append(resp.Answer, &dns.A{
+			//check if the query type is  A for IPv4 or ANY
+			ARecord := &dns.A{
```

Comment:
```suggestion
			aRecord := &dns.A{
```

Generally local vars would be lowercase, even when they are acronyms or short forms I think.